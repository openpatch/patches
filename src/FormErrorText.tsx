import { ReactNode } from "react";
import { css } from "@emotion/react";
import { Text } from "./Text";
import { ExclamationCircle } from "./icons/solid";
import { Icon } from "./Icon";
import { Box } from "./Box";

export type FormErrorTextProps = {
  children?: ReactNode;
};

export const FormErrorText = ({ children }: FormErrorTextProps) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      css={(theme) => css`
        svg {
          margin-right: ${theme.space.xxsmall};
        }
      `}
    >
      <Icon color="error" size="small">
        <ExclamationCircle />
      </Icon>
      <Text textColor="error.500" role="alert">
        {children}
      </Text>
    </Box>
  );
};
