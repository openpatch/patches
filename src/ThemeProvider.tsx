import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
