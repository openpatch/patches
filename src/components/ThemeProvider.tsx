import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
