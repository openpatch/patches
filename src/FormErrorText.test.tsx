
import { render } from "@testing-library/react";
import { FormErrorText, FormErrorTextProps } from "./FormErrorText";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FormErrorTextProps) =>
render(
    <ThemeProvider>
        <FormErrorText {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});