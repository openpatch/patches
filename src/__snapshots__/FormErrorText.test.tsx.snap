// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  margin-right: 4px;
}

.emotion-2 {
  position: relative;
  border-radius: 9999px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  color: #A61B1B;
  width: 24px;
  height: 24px;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

.emotion-2 svg .primary {
  fill: #E66A6A;
}

.emotion-2 svg .secondary {
  fill: #A61B1B;
}

.emotion-3 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  color: #BA2525;
}

<div>
  <div
    class="emotion-0 emotion-1"
    display="flex"
  >
    <span
      class="emotion-2"
    >
      <svg
        fill="currentColor"
        viewBox="0 0 20 20"
      >
        <path
          clip-rule="evenodd"
          d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="emotion-3 emotion-1"
      role="alert"
    />
  </div>
</div>
`;
